src/System.Private.CoreLib/src/System/Threading/WaitSubsystem.WaitableObject.Unix.cs

```diff
@@ -352,14 +350,7 @@ public bool Wait_Locked(ThreadWaitInfo waitInfo, int timeoutMilliseconds, bool i
                                 waitableObject.AcceptSignal(waitInfo);
                                 if (isAbandoned)
                                 {
-                                    if (waitHandlesForAbandon == null)
-                                    {
-                                        throw new AbandonedMutexException();
-                                    }
-                                    else
-                                    {
-                                        throw new AbandonedMutexException(i, waitHandlesForAbandon[i]);
-                                    }
+                                    return WaitHandle.WaitAbandoned + i;
```

Comment:
Now some of the `WaitSubsystem` methods return `WaitHandle.WaitAbandoned + i` and others throw `AbandonedMutexException`. Obviously that's not ideal and I will address it.